packages/core/strapi/lib/commands/utils/commander.js

```diff
@@ -37,13 +87,110 @@ const promptEncryptionKey = async (thisCommand) => {
       ]);
       opts.key = answers.key;
     } catch (e) {
-      console.error('Failed to get encryption key');
-      process.exit(1);
+      return exitWith(1, 'Failed to get encryption key');
     }
     if (!opts.key) {
-      console.error('Failed to get encryption key');
-      process.exit(1);
+      return exitWith(1, 'Failed to get encryption key');
+    }
+  }
+};
+
+/**
+ * prompt for auth information for Strapi
+ */
+const promptAuth = async (name, options) => {
```

Comment:
Still not a big fan of proposing the email/password auth for this as it should directly be something that we integrate into the CLI at a lower lever + add SSO login capabilities. But TBD with @marcoautiero.

Reply:
> * In the end, does it make sense to stop the transfer process if the user has lost their privileges after the init? Especially if we're checking every minute, a lot of data will be transferred in this timespan. Your idea of rechecking permissions on user or role changes might actually work, we could investigate this.
IMO it's necessary because theoretically the websocket can be kept alive forever. If we don't re-check, then the only way to revoke the permission would be to restart Strapi (assuming it could even be known they were still connected, and we don't have a mechanism to display that)

> * Maybe we could extract all this permission logic stuff in a dedicated utility in another file? This way we abstract the auth logic from the main logic (dedicated to the remote protocol). What do you think?
Yeah, that's why it's in the generic commander utils and only called in transfer as a hook. Eventually THIS hook will only be used by a 'login' command, and then any other commands that need auth will use a `check if logged in and load session` hook.

I'm fine breaking it off into an auth-specific file though.